ee/be/modules/350-node-local-dns/images/stale-sockets-cleaner/main.go

```diff
@@ -0,0 +1,278 @@
+	sizeofSocketRequest        = sizeofSocketID + 0x8
+	SOCK_DESTROY               = 21
+	familyIPv4                 = syscall.AF_INET
+	protoUDP                   = unix.IPPROTO_UDP
+)
+
+var (
+	native       = nl.NativeEndian()
+	networkOrder = binary.BigEndian
+)
+
+func main() {
+	log.Infof("[StaleSockCleaner] Start")
+
+	// Init kubeClient
+	config, _ := rest.InClusterConfig()
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("[StaleSockCleaner] Failed to init kubeClient. Error: %v", err)
+	}
+	log.Infof("[StaleSockCleaner] kubeClient successfully inited")
+
+	// Get name of node
+	currentNodeName := os.Getenv("NODE_NAME")
+	if len(currentNodeName) == 0 {
+		log.Fatalf("[StaleSockCleaner] Failed to get env NODE_NAME.")
+	}
+	log.Infof("[StaleSockCleaner] The current node name is %s", currentNodeName)
+
+	// Get podCIDR of the node
+	podCIDROnSameNode, err := getPodCIDR(kubeClient, currentNodeName)
+	if err != nil {
+		log.Fatalf("[StaleSockCleaner] Failed to get PodCIDR of the node. Error: %v", err)
+	}
+	log.Infof(
+		"[StaleSockCleaner] podCIDR on node %s is %s",
+		currentNodeName,
+		podCIDROnSameNode.String(),
+	)
+
+	// Main loop
+	for {
+		time.Sleep(scanInterval)
+		// Get ip of pod node-local-dns running on the node
+		nldPodNameOnSameNode, nldPodIPOnSameNode, err := getNLDPodNameIP(kubeClient, currentNodeName)
+		if err != nil {
+			log.Errorf("[StaleSockCleaner] Failed to get IP of the nld Pod. Error: %v", err)
+			continue
+		}
+		if nldPodIPOnSameNode == nil {
+			continue
+		}
+
+		// Get all UDP sockets on node
+		allUDPSockets, err := netlink.SocketDiagUDP(familyIPv4)
+		if err != nil {
+			log.Errorf("[StaleSockCleaner] Failed get UPD sockets. Error: %v", err)
+		}
+
+		/*
+			For each socket check:
+			- If DST Port is equal to nldDstPort?
+			- Is DST IP contained in podCIDR?
+			- Isn't DST IP equal to nldIP?
+			If all checks are true, then delete such socket
+		*/
+		for _, sock := range allUDPSockets {
+			if sock.ID.DestinationPort == nldDstPort &&
+				podCIDROnSameNode.Contains(sock.ID.Destination) {
+				if !sock.ID.Destination.Equal(nldPodIPOnSameNode) {
+					log.Infof(
+						"[StaleSockCleaner] Finded socket %s:%v -> %s:%v, where dst_ip is contained in the podCIDR (%s) and dst_port is equal %v.",
+						sock.ID.Source.String(),
+						sock.ID.SourcePort,
+						sock.ID.Destination.String(),
+						sock.ID.DestinationPort,
+						podCIDROnSameNode.String(),
+						nldDstPort,
+					)
+					log.Infof(
+						"[StaleSockCleaner] Pod %s has ip %s. dst ip from socket(%s) is not equal to the ip of pod. So this socket will be destroed.",
+						nldPodNameOnSameNode,
+						nldPodIPOnSameNode.String(),
+						sock.ID.Destination.String(),
+					)
+					err := destroySocket(sock.ID)
+					if err != nil {
+						if errors.Is(err, unix.EOPNOTSUPP) {
+							log.Fatalf("[StaleSockCleaner] Failed to destroy the socket becase this is not supported by underlying kernel. Error: %v", err)
```

Comment:
```suggestion
							log.Fatalf("[StaleSockCleaner] Failed to destroy the socket because this is not supported by underlying kernel. Error: %v", err)
```